framework/selftest.cpp

```diff
@@ -234,13 +234,13 @@ static int selftest_logerror_init(struct test *test)
 static int selftest_skipmsg_success_cleanup(struct test *test)
 {
     log_skip(SelftestSkipCategory, "SUCCESS after skipmsg from cleanup");
-    return EXIT_SKIP;
+    return EXIT_SUCCESS;
 }
 
 static int selftest_skipmsg_skip_cleanup(struct test *test)
 {
     log_skip(SelftestSkipCategory, "SKIP after skipmsg from cleanup");
-    return EXIT_SKIP;
+    return EXIT_SUCCESS;
```

Comment:
Doesn't it beat the purpose of the test (at least the message in `log_skip()`?

Reply:
`log_skip` should mark it as having skipped, regardless of what you return, especially because you may forget or not all code paths may get it. That's the purpose of this change.

Reply:
I am sorry, I was making the comment on the run. What I mean is the test declares to test `EXIT_SKIP` after `log_skip()` (in the func signature, in the `log_skip()` message too). With this change it changes the nature of the test completely.